### DISCUSSION TITLE
Better UI/UX design for the RustDesk desktop client main window

### DISCUSSION THREAD

**Discussion by rso42** [rustdesk/rustdesk#7097] [Ideas]
I've been using RustDesk just for a few days now.
I've used all kind of other Remote Control tools over the past 20 years.

### The RustDesk desktop client UI just feels no there yet.
There are many things that have caught my eye.
I don't know if anyone here is interested in this
I'm writing here while my first impression is still fresh.
And becuse **I really like the product and I would love to see it become even better.**

### To stimulate the discussion in terms of UI/UX, here is a small mockup of a different approach

![RustDesk-Client-alt-UI-v0 1](https://github.com/rustdesk/rustdesk/assets/23501319/06e8f566-f34f-4553-8163-ee7acf99a9b2)

### My main criticisms of the current client UI are:

1. Less intuitive layout. Control elements arranged randomly. Shifting frames, box sizes, areas with background color, different button styles, icon styles.
=> I have noticed that new users find it more difficult to install and set up the RustDesk client than I am used to when I am on the phone.

2. No clear hierarchy in attention management
=> The ID and one-time password are unnecessarily well hidden between a lot of other text. Sounds stupid, but if someone is having problems on the computer and is upset and I'm supposed to help, the ID and password could be even clearer.

3. ID server name hidden in settings window
=> Using the same ID server is the basis for the whole program. Why is this important setting so well hidden? If the person opposite me has a RustDesk client previously installed by someone else, I would first have to ask the user by phone to look up the ID server in the settings.

4. No clear branding. Logo missing.
=> I think RustDesk works very well technically. But new customers have to take my word for it. The program itself doesn't look trustworthy enough yet.

5. Dead face smiley in right area on newly installed clients (with no connection history yet)
=> Far... what?. Why would anyone put such a negative and scary image into a software whose sole purpose is to be trustworthy? IMO it would be much stronger if positive safety aspects were emphasized. Also with a clearer design and without smileys.

**Comment by rustdesk** [comment]
@grummbeer

**Comment by rustdesk** [comment]
Thank you very much.
Our design is not good enough, but honestly, your mockup looks worse for me. :(
UI is something every one can have an opinion.

**Comment by rso42** [comment]
I know. The design won't win any prizes.
But I've learned you have to simply visualize things to get a discusstion about design started. That was important to me.
Thank you for your quick and open response!

Maybe the first step for imporvement could be to discuss what we don't like.

If you hope to gain new insights from a discussion, just write your thoughts on what you don't like about the current client or the mockup, whatever.

I will be happy to contribute my thoughts and ideas.

**Comment by grummbeer** [comment]
Starting this discussion is necessary and a good idea. 

> 1. Less intuitive layout. Control elements arranged randomly. Shifting frames, box sizes, areas with background color, different button styles, icon styles.

Yes, I think that should be the consensus.

The UI has grown as needed. Things were added because they were needed, but without specific guidelines or templates for this or that particular element. 

> I have noticed that new users find it more difficult to install and set up the RustDesk client than I am used to when I am on the phone.

One interesting point here (in my opinion) is, that reaching the settings is not clear. Your mock shows this "gear" icon what is currently missing. The tiny "hamburger" icon at the top right is not obvious enough and at the wrong position. Its Part of "1."

I have to think a bit first and look again at what I like or don't like.

**Comment by rso42** [comment]
For me, an evolved user interface is often better than one designed from scratch. Especially with technical tools.
The ability to move and evolve quickly is an important aspect in the beginning.
I would keep that in mind when actively optimizing the user interface now.
I like to reserve areas in the user interface where developers can add new features quickly and easily.
That way you can focus on functionality without worrying too much about the user interface.

But as the user base grows, so does the responsibility not to break daily habits when making changes to the user interface.
You never want to create a second old.reddit conflict with your user base.

That's why I think the next step might be consolidation rather than a complete redesign.
Keep the good things, maybe organize them better.
Improve the bad things. Or get rid of them.
Make room for further development.
Plan a few steps ahead, but take it one step at a time.
Make changes comprehensible.
And listen to what the users say.

**Comment by xlionjuan** [comment]
not bad idea, but the colors in the demo isn't very great :3

**Comment by GerdBernau** [comment]
I think point 3 is very important. More and more companies are using Rustdesk and using their own server. The ID server is then simply exchanged by other users and you don't notice it immediately when you want to establish a connection.

**Comment by rso42** [comment]
Here is another mockup.
It's neither complete nor perfect.
But I think it shows some interesting ideas and improvements.

![RustDeskClient-altUI-home-v0 2](https://github.com/rustdesk/rustdesk/assets/23501319/45406930-17e2-47cd-84e2-cab7dc13d9a6)
_home view_

![RustDeskClient-altUI-remotecontrol-v0 2](https://github.com/rustdesk/rustdesk/assets/23501319/b8306f1c-c269-4208-bbb1-d5aa402ec02a)
_remote control view_


### ideas and improvements
- The home view should be as simple and self-explanatory as possible. the target group is the end user (stressed, in trouble, inexperienced) whose pc is to be controlled.
- The remote control view should be a power tool for everyday use by IT professionals.
- There should be a new feature 'client groups'. Each client group should have their own ID server settings. and you need to be able to quickly switch beween those client gropus.
- The new client groups solve the dilemma caused by the fact that everyone hosts their own ID server and therefore power users may have to switch ID server settings frequently in future. While new users will be on the default rustdesk ID server.
- Switching beween client groups, you could control different clients via different ID servers at the same time from your one RustDesk app.
- Next to the tabs in the window title bar there should be an input field to quickly start a connection via your default ID server.
- The RustDesk logo should be clearly visible in the home view (perhaps this would also be a good place to add your own logo to the app later).
- The remote control view should have a better list view. And it doesn't have to have a fancy grid view with random colors.
- The small colored dot should only be used throughout the app for the server connection status. Tags should not have a similar looking small dot.

**Reply by rustdesk** [reply]
Thank you very much. The layout get better now, but the colors seem not good for me. :(

**Comment by misiek4** [comment]
I agree with rso42. In the current client if you have 200 devices, managing them (scrolling) is a nightmare.

**Comment by rustdesk** [comment]
Pinned for more discussions

New UI and classic UI can coexist, let users choose by themself. We encourage enthusiasts work on the new UI, it needs not to be perfect at the beginning, let's improve gradually.

**Reply by rso42** [reply]
Thank you for your attention to this discussion. I totally agree with your suggestion to gradually improve a new UI.

**Comment by rso42** [comment]
And here's another mockup.
I changed (and reduced) some colors.
The mockup also includes improvements in many details.
For me, it's starting to feel like a program that I would like to use regularly.


![RustDeskClient-altUI-home-v0 3](https://github.com/rustdesk/rustdesk/assets/23501319/32eab397-677c-4e12-9bfc-9c16fb64121f)
_home view_

![RustDeskClient-altUI-remotecontrol-v0 3](https://github.com/rustdesk/rustdesk/assets/23501319/e2484168-3a36-416e-820f-9e99e1ad6f60)
_remote control view_

**Reply by dgva** [reply]
Thank you for this contribution, I also love RustDesk (been using it for about a year) and want to see it grow into a more mature product overall. 

The colors seem too much "in your face", I would rather see more neutral backgrounds and see the strong blue as an accent. I tend to prefer UI done like that, overall. But the reinforcement of the message to not give sensitive data is very important, I like it

The Remote Control View is just perfect, all the information with a single glance! This is something I'm sure all users can agree on. Folder/tags to separate many connections, and the management of connections and even the active connetions seems more intuitive like this.

Great ideias overall, very glad to see this effort!

**Comment by george-martinec** [comment]
This UI sucks dude, color contrast is off, text are not in golden ratio, etc. current UI is better, not awesome but better.

for starters you should use their primary blue color and white to have same color palette as their website, use black text for primary text and less black for accent text to help users to not strain eyes, etc.
